bciers/apps/administration/app/components/operators/OperatorForm.tsx

```diff
@@ -0,0 +1,82 @@
+"use client";
+
+import { useState } from "react";
+import { useParams, useRouter } from "next/navigation";
+import SingleStepTaskListForm from "@bciers/components/form/SingleStepTaskListForm";
+import { RJSFSchema } from "@rjsf/utils";
+import { actionHandler } from "@bciers/actions";
+import { operatorUiSchema } from "../../data/jsonSchema/operator";
+import Snackbar from "@mui/material/Snackbar";
+import { GREEN_SNACKBAR_COLOR } from "@bciers/styles";
+
+export interface OperatorFormData {
+  [key: string]: any;
+}
+
+interface Props {
+  schema: RJSFSchema;
+  formData: OperatorFormData;
+  isCreating?: boolean;
+}
+
+export default function OperatorForm({
+  formData,
+  schema,
+  isCreating,
+}: Readonly<Props>) {
+  // @ ts-ignore
+  const [error, setError] = useState(undefined);
+  const [isSuccessful, setIsSuccessful] = useState(false);
+  const router = useRouter();
+  const params = useParams();
+  return (
+    <>
+      <Snackbar
+        open={isSuccessful}
+        message="Your edits were saved successfully"
+        autoHideDuration={5000}
+        sx={{
+          "& .MuiSnackbarContent-root": {
+            backgroundColor: GREEN_SNACKBAR_COLOR,
+          },
+        }}
+      />
+
+      <SingleStepTaskListForm
+        error={error}
+        schema={schema}
+        uiSchema={operatorUiSchema}
+        formData={formData}
+        onSubmit={async (data: { formData?: any }) => {
+          const method = isCreating ? "POST" : "PUT";
+          const endpoint = isCreating
+            ? "registration/v2/user-operators"
+            : `registration/v2/user-operators/current`;
+          const pathToRevalidate = isCreating ? "dashboard/facilities" : `tbd`;
+          const response = await actionHandler(
+            endpoint,
+            method,
+            pathToRevalidate,
+            {
+              body: JSON.stringify(data.formData),
+            },
+          );
+          if (response.error) {
+            setError(response.error);
+            // return error so SingleStepTaskList can re-enable the submit button and user can attempt to submit again
+            return { error: response.error };
+          }
+          setIsSuccessful(true);
+          if (isCreating) {
+            router.replace(
+              `/operations/${params.operationId}/facilities/${response.id}?title=${response.name}`,
```

Comment:
` router.replace(/operations/${params.operationId}/facilities/${response.id}?title=${response.name},` will need to be modified to follow the new breadcrumb pattern:

The pattern for the query string parameter should follow the format `{name-of-preceding-folder}Title`. For example, `?operationsTitle`

Reply:
I copy/pasted this component from facilities and forgot to update all the create stuff with tbd